cudax/include/cuda/experimental/__device/device_vector.cuh

```diff
@@ -0,0 +1,231 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__DEVICE_DEVICES_RANGE
+#define _CUDAX__DEVICE_DEVICES_RANGE
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__cccl/attributes.h>
+#include <cuda/std/__cuda/api_wrapper.h>
+#include <cuda/std/cassert> // IWYU pragma: keep (for assert)
+#include <cuda/std/iterator> // IWYU pragma: keep (for random_access_iterator_tag)
+
+#include <cuda/experimental/__device/device.cuh>
+
+namespace cuda::experimental
+{
+namespace __detail
+{
+//! @brief A range of all available CUDA devices
+class device_vector
```

Comment:
That name is bad, we want this to be a proper container at some time and not a storage for devices

Maybe name it `devices` or `device_list`